### PR TITLE
[Core] Adding CopySubModelPartStructure

### PR DESCRIPTION
**Pull request by loumalouomega**
**Description**
 Adding CopySubModelPartStructure

**Changelog**
- Adding CopySubModelPartStructure
- Adding CopySubModelPartStructure test

### CODE REVIEW

File: kratos/utilities/auxiliar_model_part_utilities.h [hunk 1]
```diff
@@ -581,6 +587,16 @@ class KRATOS_API(KRATOS_CORE) AuxiliarModelPartUtilities
     ///@name Private Operations
     ///@{
 
+    static inline void AuxiliarCopySubModelPartStructure(ModelPart& rModelPart, const ModelPart& rModelPartToCopy)
```
Comment (pooyan-dadvand) [inline kratos/utilities/auxiliar_model_part_utilities.h]: A recursive function can't be inlined. (It is ignored)

File: kratos/utilities/auxiliar_model_part_utilities.cpp [hunk 2]
```diff
@@ -24,6 +24,14 @@
 
 namespace Kratos
 {
+void AuxiliarModelPartUtilities::CopySubModelPartStructure(const ModelPart& rModelPartToCopy)
+{
+    AuxiliarCopySubModelPartStructure(mrModelPart, rModelPartToCopy);
```
Comment (pooyan-dadvand) [inline kratos/utilities/auxiliar_model_part_utilities.cpp]: I think  in this case the `AuxiliarCopySubModelPartStructure` with two arguments is a more clear interface than the `CopySubModelPartStructure` 

At least I would expose both!

File: kratos/utilities/auxiliar_model_part_utilities.h [hunk 3]
```diff
@@ -101,6 +101,12 @@ class KRATOS_API(KRATOS_CORE) AuxiliarModelPartUtilities
     ///@name Operations
     ///@{
 
+    /**
+     * @brief This method copies the structure of submodelparts
+     * @param rModelPartToCopy The model part to copy
+     */
+    void CopySubModelPartStructure(const ModelPart& rModelPartToCopy);
```
Comment (pooyan-dadvand) [inline kratos/utilities/auxiliar_model_part_utilities.h]: `rModelPartToCopyFrom`

File: kratos/utilities/auxiliar_model_part_utilities.h (hunk 1 above)
Comment (pooyan-dadvand) [reply kratos/utilities/auxiliar_model_part_utilities.h]: I would also chage the arguments order to (from, to)

File: kratos/utilities/auxiliar_model_part_utilities.cpp (hunk 2 above)
Comment (loumalouomega) [reply kratos/utilities/auxiliar_model_part_utilities.cpp]: Thi thing is that AuxiliarModelPartUtilities have a modelpart as input..., I can do an exception here...

**Comment by loumalouomega** [comment]
Ping @pooyan-dadvand

**Comment by pooyan-dadvand** [comment]
@loumalouomega the name `rModelPartToCopy` is very confusing for me and the method comment doesn't help. Does it mean "to copy into it" or "to copy from it". 

Clearly, English is not my native language!! 😜

**Comment by loumalouomega** [comment]
> @loumalouomega the name `rModelPartToCopy` is very confusing for me and the method comment doesn't help. Does it mean "to copy into it" or "to copy from it".
> 
> Clearly, English is not my native language!! stuck_out_tongue_winking_eye

Now?

**Comment by loumalouomega** [comment]
> > @loumalouomega the name `rModelPartToCopy` is very confusing for me and the method comment doesn't help. Does it mean "to copy into it" or "to copy from it".
> > Clearly, English is not my native language!! stuck_out_tongue_winking_eye
> 
> Now?

Ping

**Comment by pooyan-dadvand** [comment]
So why do you have the argument list as (to, from) instead of (from, to)?

**Comment by loumalouomega** [comment]
> So why do you have the argument list as (to, from) instead of (from, to)?

I thought this was the final decision?

**Comment by loumalouomega** [comment]
> So why do you have the argument list as (to, from) instead of (from, to)?

Updated

**Review by pooyan-dadvand — APPROVED** [review]
Thanks!

**Pull request merged**